scripts/snakemake_sophie_get_stats_DESeq.R

```diff
@@ -35,7 +35,20 @@ get_DE_stats_DESeq <- function(grp_metadata_file,
   }
  
   metadata$group <- as.factor(metadata$group)
-  ddset <- DESeqDataSetFromMatrix(expression_data, colData=metadata, design = ~group)
+  
+  # parse the grp_metadata_file character string to determine which deseq2 comparison we're performing
+  comparison <- gsub(".*__", "", basename(grp_metadata_file))
+  comparison <- gsub("_groups.tsv", "", comparison)
+  
+  if(comparison == "spu-vs-spu_m"){
+    # control for individual when comparing spu vs spu_m.
+    metadata$subject_id <- gsub("_M", "", metadata$sample)
+    ddset <- DESeqDataSetFromMatrix(expression_data, colData=metadata, 
```

Comment:
```suggestion
    ddset <- DESeqDataSetFromMatrix(expression_data, colData = metadata, 
```
I'm guessing you added spaces around the `design =` on the next line and not here because it's a more complex expression - personally, though, I'd probably just suggest picking one (spaces or no spaces) and sticking with it. Looks like the [tidyverse style guide](https://style.tidyverse.org/syntax.html#infix-operators) suggests adding spaces, so I'd default to that if you don't have a strong opinion.

Reply:
thank you!